packages/react-ui-components/src/IconButton/iconButton.tsx

```diff
@@ -18,6 +18,11 @@ export interface IconButtonProps extends HTMLButtonElementAttributesExceptStyle
      */
     readonly icon: string | IconName;
 
+    /**
+     * Additional props passed to the icon
+     */
+    readonly iconProps?: React.ComponentProps<typeof Icon>;
```

Comment:
we discussed whether to use slots or additionalProps but it seems to boil down to personal preference - and i dont know if we decided in the ui already for the one or other pattern

im team "slots" but this is also fine with me (as its not such a critical change)